### PR TITLE
Add a custom header with timezone offset to a CreateClaim mutation

### PR DESCRIPTION
**Pull request by fd-pliska**
# Jira Issue: [IP-59](https://hedvig.atlassian.net/browse/IP-59?atlOrigin=eyJpIjoiNmM4ZTRmNmVmNWQxNDYyNTg2ODlkNjZjM2I1NGY0ZjAiLCJwIjoiaiJ9)

## What?
- Add a custom header to CreateClaim mutation to pass timezone offset to the backend

## Why?
- To ensure that the claim's registration date is always saved in UTC

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [ ] Codescouted
- [ ] Unit tests written
- [x] Tested locally

### CODE REVIEW

File: src/server/apollo-client.ts [hunk 1]
```diff
@@ -36,6 +37,21 @@ const getItemWithExpiry = (key) => {
...
+
+  return {
+    ...previousContext,
+    headers: {
+      ...headers,
+      clientTzOffset: new Date().getTimezoneOffset(),
+    },
+  }
+})
+
```
Comment (maxi3336) [inline src/server/apollo-client.ts:54]: Looks perfect! 🤩

**Review by rasmusguterstam — APPROVED** [review]
image

File: src/server/apollo-client.ts [hunk 2]
```diff
@@ -36,6 +37,21 @@ const getItemWithExpiry = (key) => {
   return item.value
 }
 
+const addTimezoneOffsetHeader = setContext((_operation, previousContext) => {
+  const { headers, clientTzOffset } = previousContext
+  if (!clientTzOffset) {
+    return previousContext
+  }
```
Comment (vonElfvin) [inline src/server/apollo-client.ts:44]: Can't we just always include this in all requests? I see no risk in including it regardless of reqest

File: src/server/apollo-client.ts [hunk 3]
```diff
@@ -36,6 +37,21 @@ const getItemWithExpiry = (key) => {
...
+
+  return {
+    ...previousContext,
+    headers: {
+      ...headers,
+      clientTzOffset: new Date().getTimezoneOffset(),
+    },
+  }
+})
```
Comment (vonElfvin) [inline src/server/apollo-client.ts:53]: I'd say we should always include it, and rename it to `clientTimezoneOffset`
```suggestion
const addTimezoneOffsetHeader = setContext((_operation, previousContext) => {
  return {
    ...previousContext,
    headers: {
      ...headers,
      clientTimezoneOffset: new Date().getTimezoneOffset(),
    },
  }
})
```

File: src/types/generated/graphql.tsx [hunk 4]
```diff
@@ -6313,7 +6313,11 @@ export function useCreateClaimMutation(
     CreateClaimMutationVariables
   >,
 ) {
-  const options = { ...defaultOptions, ...baseOptions }
+  const options = {
+    ...defaultOptions,
+    ...baseOptions,
+    context: { clientTzOffset: -60 },
```
Comment (vonElfvin) [inline src/types/generated/graphql.tsx:6319]: Wait did you manually update the `generated/graphql.tsx` file? 🤔

**Review by vonElfvin — COMMENTED** [review]
Very nice! Really awesome work 🙇 

I added some things I'd like to discuss 🙏

File: src/types/generated/graphql.tsx (hunk 4 above)
Comment (fd-pliska) [reply src/types/generated/graphql.tsx:6319]: Oh shoot, I guess I did, but I didn't even notice what I did

**Pull request merged**